node/circuits.js

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+"use strict";
+
+var inherits = require('util').inherits;
+var States = require('./states');
+var EventEmitter = require('./lib/event_emitter');
+var StateMachine = require('./state_machine');
+var errors = require('./errors');
+
+// The each circuit uses the circuits collection as the "nextHandler" for
+// "shouldRequest" to consult.  Peers use this hook to weight peers both by
+// healthy and other factors, but the circuit only needs to know about health
+// before forwarding.
+
+function AlwaysShouldRequestHandler() { }
+
+AlwaysShouldRequestHandler.prototype.shouldRequest = function shouldRequest() {
+    return true;
+};
+
+var alwaysShouldRequestHandler = new AlwaysShouldRequestHandler();
+
+//  circuit = circuits
+//      .circuitsByServiceName[serviceName]
```

Comment:
add a second column comment noting each class type at each level ... would've really helped explain things first time I had read this... plus a comment above saying: "Here's how the following classes related:"